### DISCUSSION TITLE
Argument() missing 1 required positional argument: "default"

### DISCUSSION THREAD

**Discussion by abdelkareemkobo** [fastapi/typer#656] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer
from typing_extensions import Annotated

def main(
    name:Annotated[
        str,
        typer.Argument(
            help="Who to greet",show_default="Deadpoolio the amazing's name"
        ),
    ]="Wade Wilson"):
    print(f"Hello {name}")

if __name__=="__main__":
    typer.run(main)
```

### Description

This code must work and when i typed python main.py --help   or python main.py it should run but i get the following error:
Argument() missing 1 required positional argument: "default" 

### Operating System

Linux

### Operating System Details

pop os 

### Typer Version

0.7.0

### Python Version

3.10.11

**Comment by Tong-Du** [comment, marked as answer]
Works fine on typer 0.9.0
